app/assets/javascripts/modules/slotpicker/moj.datepicker.js

```diff
@@ -0,0 +1,115 @@
+(function() {
+  'use strict';
+
+  moj.Modules.Slotpicker = moj.Modules.Slotpicker || {}
+
+  var Datepicker = function(el, options) {
+
+    this.$el = el;
+
+    this.init = function() {
+      moj.Modules.Slotpicker.SlotSource.init();
+      this.previouslySelectedSlot = moj.Modules.Slotpicker.SlotSource.getValue();
+      var availableSlots = moj.Modules.Slotpicker.SlotSource.getAvailableSlots(),
+          firstDate = availableSlots[0].date.split('-');
+
+      this.defaults = {
+        curYear: firstDate[0],
+        year: firstDate[0],
+        curMonth: new Date(firstDate[0], parseInt(firstDate[1], 10) - 1, firstDate[2]).getMonth(),
+        month: new Date(firstDate[0], parseInt(firstDate[1], 10) - 1, firstDate[2]).getMonth(),
+        currentDate: false,
+        availableSlots: availableSlots
+      }
+
+      this._settings = $.extend({}, this.defaults, options);
+      this.setup();
+    };
+
+    return this;
+
+  };
+
+  Datepicker.prototype = {
+
+    cacheEls: function() {
+      this.$submitBtn = $('#' + this._settings.slotSubmit);
+      this.$cancelBtn = $('#' + this._settings.slotCancel);
+      this.$deleteBtn = $('#' + this._settings.slotDelete);
+      this.$skipBtn = $('#' + this._settings.slotSkip);
+    },
+
+    bindEvents: function() {
+      var self = this;
+
+      this.$cancelBtn.on('click', function(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        self.updateSlotSource(self.previouslySelectedSlot);
+        var form = $(this).parents('form');
+        form.submit();
+      });
+
+      this.$deleteBtn.on('click', function(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        moj.Modules.Slotpicker.SlotSource.removeSlot();
+        var form = $(this).parents('form');
+        form.submit();
+      });
+
+      this.$skipBtn.on('click', function(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var form = $(this).parents('form');
+        $('#slots_step_skip_remaining_slots').val(true);
```

Comment:
Could this be an `this.$myId`?

Reply:
Again, this is an element that is referenced once so doesn't need to be cached.